pkg/volume/csi/csi_plugin.go

```diff
@@ -659,10 +658,9 @@ func (p *csiPlugin) NewBlockVolumeMapper(spec *volume.Spec, podRef *api.Pod, opt
 	}
 
 	if err := saveVolumeData(dataDir, volDataFileName, volData); err != nil {
-		if removeErr := os.RemoveAll(dataDir); removeErr != nil {
-			klog.Error(log("failed to remove dir after error [%s]: %v", dataDir, removeErr))
-		}
-		return nil, errors.New(log("failed to save volume info data: %v", err))
+		errMsg := log("failed to save volume info data: %v", err)
+		klog.Error(errMsg)
+		return nil, errors.New(errMsg)
```

Comment:
sorry for taking time on this but rather than dropping this check altogether can we simple call the implementation of `removeMountDir` which should be relatively safe because it only removes data file and then removes with directory without ever calling `os.RemoveAll` ?

Reply:
RemoveAll may delete data in the mountpoint, and it is unexpected.  I think it is safe to call removeMountDir here because removeMountDir won't delete data.

Reply:
Yes - can you update this PR to use `removeMountDir` ?

Reply:
Sure, I will update the code and do a further test.